lib/src/model.dart

```diff
@@ -1268,8 +1271,46 @@ abstract class SourceCodeMixin {
     }).join('\n');
   }
 
+  String get crossdartHtmlTag {
+    if (_crossdartUrl != null) {
+      return "<a class='crossdart' href='${_crossdartUrl}'>Link to Crossdart</a>";
+    } else {
+      return "";
+    }
+  }
+
+  int get _lineNumber {
+    var node = element.computeNode();
+    if (node is Declaration && (node as Declaration).element != null) {
+      return cache.lineNumber((node as Declaration).element.source.fullName, node.offset);
+    } else {
+      return null;
+    }
+  }
+
+  String get _crossdartUrl {
+    if (_lineNumber != null && _sourceFilePath != null) {
+      String packageName = library.package.isSdk ? "sdk" : library.package.name;
+      String packageVersion = library.package.version;
+      String url = path.join("http://www.crossdart.info/p", packageName, packageVersion, _sourceFilePath) + ".html";
```

Comment:
Unfortunately, google cloud storage doesn't allow for custom certs + custom DNS :(

You could put cloudfront in front.